tests/rust_tests.rs

```diff
@@ -367,8 +555,461 @@ fn mint_nft_ft_test() {
+                    .get(),
+                whitelist_enabled: sc.white_list_enabled().get(),
+                is_whitelisted: sc
+                    .white_list()
+                    .contains(&managed_address!(first_user_address)),
+            };
+            assert_eq!(
+                sc.get_user_data_out(
+                    &managed_address!(first_user_address),
+                    &managed_token_id_wrapped!(TOKEN_ID)
+                ),
+                data_out
+            );
+        })
+        .assert_ok();
+}
+
+#[test] //Tests the whitelist functionality
+fn white_list_test() {
+    let mut setup = setup_contract(datanftmint::contract_obj);
+    let b_wrapper = &mut setup.blockchain_wrapper;
+    let owner_address = &setup.owner_address;
+    let first_user_address = &setup.first_user_address;
+
+    b_wrapper
+        .execute_query(&setup.contract_wrapper, |sc| {
+            let whitelist = MultiValueEncoded::new();
+            sc.set_whitelist_spots(whitelist)
+        })
+        .assert_user_error("Given whitelist is empty");
+
+    b_wrapper
+        .execute_tx(
+            &owner_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                let mut args = MultiValueEncoded::new();
+                args.push(managed_address!(first_user_address));
+                sc.set_whitelist_spots(args);
+            },
+        )
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &owner_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                let mut args = MultiValueEncoded::new();
+                args.push(managed_address!(first_user_address));
+                sc.set_whitelist_spots(args);
+            },
+        )
+        .assert_user_error("Address already in whitelist");
+
+    b_wrapper
+        .execute_query(&setup.contract_wrapper, |sc| {
+            let whitelist = MultiValueEncoded::new();
+            sc.remove_whitelist_spots(whitelist)
+        })
+        .assert_user_error("Given whitelist is empty");
+
+    b_wrapper
+        .execute_tx(
+            &owner_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                let mut args = MultiValueEncoded::new();
+                args.push(managed_address!(first_user_address));
+                sc.remove_whitelist_spots(args);
+            },
+        )
+        .assert_ok();
+
+    b_wrapper
+        .execute_tx(
+            &owner_address,
+            &setup.contract_wrapper,
+            &rust_biguint!(0),
+            |sc| {
+                let mut args = MultiValueEncoded::new();
+                args.push(managed_address!(first_user_address));
+                sc.remove_whitelist_spots(args);
+            },
+        )
+        .assert_user_error("Address not in whitelist");
+}
+
+#[test] // Tests burn functionality
+fn burn_token_tests() {

```

Comment:
This method is named "tests" while the others are named "test". Small detail, but generally we should try following a pattern